osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -87,14 +87,15 @@ public void Invalidate(IBeatmapInfo beatmap)
         }
 
         /// <summary>
-        /// Retrieves a bindable containing the star difficulty of a <see cref="BeatmapInfo"/> that follows the currently-selected ruleset and mods.
+        /// Retrieves a bindable containing the star difficulty of a <see cref="BeatmapInfo"/> that follows the currently-selected ruleset, along with mods if <paramref name="applyMods"/> is <c>true</c>.
         /// </summary>
         /// <param name="beatmapInfo">The <see cref="BeatmapInfo"/> to get the difficulty of.</param>
+        /// <param name="applyMods">Whether the currently-selected mods should be applied to the star difficulty calculation in this bindable.</param>
```

Comment:
Personal opinion: I don't like this param if I'm honest. Feels like too much options; to me the bindable should either follow game-global state to a T or not at all. I'd rather stick to how the previous code worked, but supply the ruleset from the game-global bindable instead.

Reply:
I don't quite understand the concern with having this parameter, but supplying the game-wide ruleset bindable is going to look pretty much ugly (one way is `OsuGame.Dependencies.Get<IBindable<RulesetInfo>>()`, another is exposing `OsuGame.Ruleset` and using it), both of which feel weird to me at face value.

Reply:
I look rather unfavorably at boolean arguments no matter the case. [And I'm hardly alone on this.](https://www.google.com/search?q=boolean%20arguments%20bad%20&ie=utf-8&oe=utf-8&client=firefox-b-m)

Here's what I'm going to ask: aside from this one hyperspecific bug caused by one component, why would you ever want the thing whose express purpose is to always follow global game state to *not fully* follow game state?

And I'm not buying the argument that alternatives are worse unless I see a full diff that's conclusively worse. I don't even get where "exposing `OsuGame.Ruleset`" is coming from - it's cached, no?

Reply:
The game-wide ruleset bindable is cached, yes. But it's overwritten by the decoupled one in song select. That's why I have to either retrieve it from the game's dependencies list or expose it directly.

But I understand your concern of boolean parameters now, fair enough. I'll expose the bindable and use it with an explanatory comment. Should do it for now.

Reply:
@frenzibyte before i dive into reviewing this (or #22032), were you planning to make the change proposed above? Or should I look at #22032 as the preferred solution first?

Reply:
Never mind.